kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/encrypt/InBandEncryptionManager.java

```diff
@@ -127,21 +131,24 @@ private CompletionStage<MemoryRecords> attemptEncrypt(@NonNull String topicName,
                     return CompletableFuture.completedFuture(encryptedMemoryRecords);
                 }
                 catch (ExhaustedDekException e) {
-                    rotateKeyContext(encryptionScheme, dek);
-                    // fall through to recursive call below...
+                    dek.rotate();
+                    // fall through to retry
+                }
+                catch (DestroyedDekException e) {
```

Comment:
the test showed up that it's possible for `Dek::encryptor` to throw DestroyedDekException, in which case we don't want to initiate the rotate, the call that exhausted the DEK should have done that. We fall through and retry.